### ISSUE TITLE
Add hint about stack name format

### ISSUE THREAD

**Issue by pbzdyl** [pulumi/pulumi#2421]
Currently when using `pulumi new` or `pulumi stack init` or when creating a new stack after `pulumi select` -> `create new stack` it is not clear what is stack name format.

I saw the recent change in the stack reference change which is now `org-name/project-name/stack-name` so I thought I should use the same format for the commands above but I got an error.

Currently pulumi cli displays this:

```
$ pulumi stack
Please choose a stack, or create a new one: <create a new stack>
Please enter your desired stack name: |
```

It would be nice to get a hint like:
```
$ pulumi stack
Please choose a stack, or create a new one: <create a new stack>
Use the following format: <org-name>/<stack-name> or <stack-name> for your non-organization account
Please enter your desired stack name: |
```

**Comment by ellismg** [comment]
@justinvp I landed an improvement here in #2446, but you said you'd like to keep this open an assigned to you as you wanted to do a little more cleanup here.

**Comment by pgavlin** [comment]
@justinvp is there more work to do here?

**Comment by justinvp** [comment]
The ideal experience is to prompt for the owner like we do on https://app.pulumi.com/site/new-project, doing so only if the user is a member of at least one organization other than their user org where the user has permissions to create stacks (otherwise, don't prompt and the stack will go under the user org).

The CLI could make use of the following service APIs to accomplish this:
 - `/api/user` to get the orgs the user is a member of (the CLI already calls this API to get the current user name)
- `/orgs/<org>/metadata` for each org to determine permissions (we assume the user will be able to create stacks in the org if `orgMetadata.userRole === "admin" || (orgMetadata.userRole === "member" && orgMetadata.membersCanCreateStacks`)

I assume we don't mind having the CLI depend on the latter API, but we should make an explicit decision. An alternative would be to provide a more optimized API that would return all orgs that the user has permission to create stacks in, in a single call (perhaps adding more information to the data returned from `/api/user`).

---

A simpler bandaid would be to do something similar to the change in #2446, and somehow indicate that the org can be specified (e.g. `owner/stack`) when prompting the user for the stack name.

**Comment by mikhailshilkov** [comment]
@justinvp I implemented the hint in #3066

I like your other suggestion about the owner prompt. I guess we should implement it consistently in all places. Let's decide on APIs. I'm not familiar with the current API, but my gut-feel preference would be the "alternative" option where the logic sits in backend.

Maybe, close this issue as the hint is done, and open a new one with owner selector?

**Comment by justinvp** [comment]
> Maybe, close this issue as the hint is done, and open a new one with owner selector?

Sounds good to me. I've opened #3076. Since it's essentially a new feature (or larger enhancement), I don't think it meets the bar for this milestone, so I've left the milestone unassigned for now.
